x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -163,20 +171,20 @@ orderExpression
     ;
 
 keepCommand
-    :  KEEP sourceIdentifier (COMMA sourceIdentifier)*
-    |  PROJECT sourceIdentifier (COMMA sourceIdentifier)*
+    :  KEEP qualifiedNamePattern (COMMA qualifiedNamePattern)*
+    |  PROJECT qualifiedNamePattern (COMMA qualifiedNamePattern)*
```

Comment:
Support nested field without quoting by using qualified names.
That is treating height.float as height - one unquoted identifier and float another unquoted identifier.